test/server/listener_socket_option_impl_test.cc

```diff
@@ -0,0 +1,85 @@
+#include "server/listener_socket_option_impl.h"
+
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/threadsafe_singleton_injector.h"
+
+#include "gtest/gtest.h"
+
+using testing::Invoke;
+using testing::Return;
+using testing::_;
+
+namespace Envoy {
+namespace Server {
+
+class ListenerSocketOptionImplTest : public testing::Test {
+public:
+  ListenerSocketOptionImplTest() { socket_.local_address_.reset(); }
+
+  NiceMock<Network::MockListenSocket> socket_;
+  Api::MockOsSysCalls os_sys_calls_;
+  TestThreadsafeSingletonInjector<Api::OsSysCallsImpl> os_calls{&os_sys_calls_};
+
+  void testSetSocketOptionSuccess(ListenerSocketOptionImpl& socket_option, int socket_level,
```

Comment:
Could this subclass `SocketOptionImplTest` and inherit this method?

Reply:
@bmetzdorf can you give me a quick yes/no on the various feedback? Thanks!

Reply:
Yes, I'm working on this one.